x-pack/plugins/index_management/public/hooks/use_details_page_mappings_model_management.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useState } from 'react';
+import { InferenceStatsResponse } from '@kbn/ml-plugin/public/application/services/ml_api_service/trained_models';
+import type { InferenceAPIConfigResponse } from '@kbn/ml-trained-models-utils';
+import { InferenceToModelIdMap } from '../application/components/mappings_editor/components/document_fields/fields';
+import { deNormalize } from '../application/components/mappings_editor/lib';
+import { useAppContext } from '../application/app_context';
+import { useComponentTemplatesContext } from '../application/components/component_templates/component_templates_context';
+import { State } from '../application/components/mappings_editor/types';
+
+interface InferenceModel {
+  data: InferenceAPIConfigResponse[];
+}
+
+type DeploymentStatusType = Record<string, 'deployed' | 'not_deployed'>;
+
+export const useDetailsPageMappingsModelManagement = (state: State) => {
+  const {
+    plugins: { ml },
+  } = useAppContext();
+
+  const { api } = useComponentTemplatesContext();
+
+  const { inferenceToModelIdMap } = state;
+
+  const [pendingDeployments, setPendingDeployments] = useState<Set<string>>(new Set());
+
+  const fetchInferenceModelsAndTrainedModelStats = useCallback(async () => {
+    const inferenceModels = (await api.getInferenceModels()) as InferenceModel;
+    const trainedModelStats = await ml?.mlApi?.trainedModels.getTrainedModelStats();
+
+    return { inferenceModels, trainedModelStats };
+  }, [api, ml]);
+
+  const getTrainedModelStats = useCallback(
+    (modelStats?: InferenceStatsResponse): DeploymentStatusType => {
+      const deploymentStatsByModelId =
+        modelStats?.trained_model_stats.reduce<DeploymentStatusType>((acc, modelStat) => {
+          if (modelStat.model_id) {
+            acc[modelStat.model_id] =
+              modelStat?.deployment_stats?.state === 'started' ? 'deployed' : 'not_deployed';
+          }
+          return acc;
+        }, {}) ?? {};
+
+      return deploymentStatsByModelId;
+    },
+    []
+  );
+
+  const getDefaultInferenceIds = useCallback((deploymentStatsByModelId: DeploymentStatusType) => {
```

Comment:
This too is a pure function that doesn't have any react context dependencies. Define it at the top level and just call it. No need to use `useCallback`, it doesn't do anything for you here.